src/ImageSharp/Formats/Jpeg/Components/Encoder/ComponentProcessor.cs

```diff
@@ -128,6 +129,18 @@ static void SumVertical(Span<float> target, Span<float> source)
                     Unsafe.Add(ref targetVectorRef, i) = Avx.Add(Unsafe.Add(ref targetVectorRef, i), Unsafe.Add(ref sourceVectorRef, i));
                 }
             }
+            else if (AdvSimd.IsSupported)
+            {
+                ref Vector128<float> targetVectorRef = ref Unsafe.As<float, Vector128<float>>(ref MemoryMarshal.GetReference(target));
+                ref Vector128<float> sourceVectorRef = ref Unsafe.As<float, Vector128<float>>(ref MemoryMarshal.GetReference(source));
+
+                // Spans are guaranteed to be multiple of 8 so no extra 'remainder' steps are needed
```

Comment:
Instead of the comment I'd prefer a Debug.Assert (or the DebugGuard).
* makes it self-documenting
* in case of failure, it crashes (with Debug builds)

Reply:
@stefannikolei if you'd replace comments with asserts - can you also replace them for Avx path please? :)

Reply:
Sure. That's was just copy pasted 😅

Reply:
We don't use `Debug.Assert`, so if anything it should be `DebugGuard`. I think in this case the comment adds a bit more info, so I wouldn't delete it even if there is an assertion.